wd_core.au3

```diff
@@ -1557,46 +1575,59 @@ EndFunc   ;==>__WD_Delete
 ; Parameters ....: $sWhere     - Name of calling routine
-;                  $i_WD_ERROR - Error constant
-;                  $sMessage   - [optional] Additional Information (default = "")
-; Return values..: Success - Error Const from $i_WD_ERROR
+;                  $iErr       - The error number from the calling function
+;                  $sMessage   - Message that will be passed to the console/output [optional]
+;                  $iExt       - Extended information from the calling function [optional]
+; Return values..: Success - $iErr
 ;                  Failure - None
-; Author ........: Thorsten Willert, Dan Pollak
-; Modified ......:
+; Author ........: Stilgar, Danp2
+; Modified ......: mLipok
 ; Remarks .......:
 ; Related .......:
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func __WD_Error($sWhere, $i_WD_ERROR, $sMessage = Default)
+Func __WD_Error($sWhere, $iErr, $sMessage = Default, $iExt = Default)
+	Local Const $sFuncName = "__WD_Error"
 	Local $sMsg
 
 	If $sMessage = Default Then $sMessage = ''
+	If $iExt = Default Then $iExt = 0
 
 	Switch $_WD_DEBUG
 		Case $_WD_DEBUG_None
 
 		Case $_WD_DEBUG_Error
-			If $i_WD_ERROR <> $_WD_ERROR_Success Then ContinueCase
+			If $iErr <> $_WD_ERROR_Success Then ContinueCase
 
 		Case $_WD_DEBUG_Info
-			$sMsg = $sWhere & " ==> " & $aWD_ERROR_DESC[$i_WD_ERROR]
-
-			If $sMessage <> "" Then
-				$sMsg = $sMsg & ": " & $sMessage
-			EndIf
-
+			Local $sExtended = (($iExt) ? (" / " & $iExt) : (""))
+			$sMsg = $sWhere & " ==> " & $aWD_ERROR_DESC[$iErr] & " ( " & $iErr & $sExtended & " )"
+			$sMsg &= (($sMessage) ? (" : " & $sMessage) : (""))
 			__WD_ConsoleWrite($sMsg)
 
-			If @Compiled Then
-				If $_WD_ERROR_MSGBOX And $i_WD_ERROR <> $_WD_ERROR_Success And $i_WD_ERROR < 6 Then MsgBox(16, "WD_Core.au3 Error:", $sMsg)
-				DllCall("kernel32.dll", "none", "OutputDebugString", "str", $sMsg)
+			If $iErr <> $_WD_ERROR_Success Then
+				If $_WD_ERROR_MSGBOX And $iErr < 6 Then
+					Local $iAnswer = MsgBox($MB_ICONERROR + $MB_OKCANCEL, "Webdriver UDF Error:", $sMsg)
+					If $iAnswer = $IDCANCEL Then
+						$iErr = $_WD_ERROR_UserAbort ; change $iErr to give a way to brake further processing by user interaction
+						If $_WD_DEBUG = $_WD_DEBUG_Info Then
+							__WD_ConsoleWrite($sFuncName & " : User Abort")
+						EndIf
+					EndIf
+				EndIf
+				If $_WD_ERROR_OUTPUTDEBUG Then
+					DllCall("kernel32.dll", "none", "OutputDebugString", "str", $sMsg)
+				EndIf
 			EndIf
+
 	EndSwitch
 
-	Return $i_WD_ERROR
+	; $iErr should be returned as "Return Value" as it is used in the calling function like this:     Return SetError(__WD_Error(...))
+	; $iExt should be returned as @extended as it can be used in the calling function like this:         Return SetError(__WD_Error(..., $iExt), @extended, ....)
+	Return SetError($iErr, $iExt, $iErr)

```

Comment:
Not sure about this, especially since you've already acknowledged in the comment two lines above about using this function as a parameter to SetError.

Reply:
Take a look on: [comments on the end](https://github.com/Danp2/au3WebDriver/pull/244/commits/88e47dfdbcf4dda93d197ad5326a67d65ecb2c92)

Reply:
My earlier comment was in reference to changing the line `	Return $i_WD_ERROR` to `Return SetError($iErr, $iExt, $iErr)`. So the issue is with adding SetError, not with your comments.

It doesn't make sense to me to add the call to SetError here when we know that it is expected that __WD_Error is being passed as a parameter to SetError at he end of each _WD* function.

Reply:
Take a look on: [Return $iErr reverted](https://github.com/Danp2/au3WebDriver/pull/244/commits/db5a376bcab3b9cb0bc8de41c149a14aa544deaf)